src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+
+// TODO: choose which parameters allowed to be tunable
+// TODO: if throws, would temp files be cleaned up automatically?
+// TODO: most robust way to check OS in method SGASerialRunner (Mac doesn't support multi-thread mechanism in SGA)?
+@CommandLineProgramProperties(
+        summary        = "Program to call SGA for performing local assembly and outputs assembled contigs.",
+        oneLineSummary = "Perform SGA-based local assembly on fasta files on Spark",
+        programGroup   = StructuralVariationSparkProgramGroup.class)
+public final class RunSGANaivelySpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc       = "Absolute path to SGA installation.",
+              shortName = "sgaPath",
+              fullName  = "fullPathToSGA",
+              optional  = false)
+    public String pathToSGA = null;
+
+    @Argument(doc       = "A path to a directory to write results to.",
+              shortName = "outDir",
+              fullName  = "outputDirectory",
+              optional  = false)
+    public String outputDir = null;
+
+    @Argument(doc       = "Number of threads to use when running sga.",
+              shortName = "t",
+              fullName  = "threads",
+              optional  = true)
+    public int threads = 1;
+
+    @Argument(doc       = "To run k-mer based read correction, filter and duplication removal in SGA or not, with default parameters.",
+              shortName = "correct",
+              fullName  = "correctNFilter",
+              optional  = true)
+    public boolean runCorrectionSteps = false;
+
+    @Override
+    public boolean requiresReads(){
+        return true;
+    }
+
+    @Override
+    public void runTool(final JavaSparkContext ctx){
+
+        final JavaRDD<GATKRead> reads = getReads();
+        // TODO: experimental interface to caller, should be getting this from JavaSparkContext
+        final JavaPairRDD<Long, Iterable<GATKRead>> readsAroundBreakpoints = reads.groupBy( read -> 1L);
+
+        final Path sgaPath = Paths.get(pathToSGA);
+
+        // map paired reads to FASTA file (temp files live in temp dir that cleans self up automatically) then feed to SGA for assembly
+        final JavaPairRDD<Long, SGAAssemblyResultType> assembledContigs = readsAroundBreakpoints.mapToPair(RunSGANaivelySpark::convertReadsToFASTQ)
+                                                                                                .mapToPair(entry -> RunSGANaivelySpark.performAssembly(entry, sgaPath, threads, runCorrectionSteps));
+
+        validateAndSaveResults(assembledContigs, outputDir);
+    }
+
+    // validates the returned result:
+    //   if all steps return 0, save the contig file and may discard the runtime information
+    //   if any step returns non-zero code, save the runtime information for that break point
+    private static void validateAndSaveResults(final JavaPairRDD<Long, SGAAssemblyResultType> results, final String outputDir){
```

Comment:
I was in a mindset that now that things don't simply fail if some potential breakpoint fails to be assembled, it should be made clear that some thing went wrong. The choice was made such that there are two object files if that's the case, and instead of mining through one file and test if it is success or not, go through the second, failure-only file and see what happened. Or simply mine the breakpoint ids in it and resubmit the job.
We can discuss this in person as it seems to be a design choice, face-to-face communication may be more efficient?